virtcontainers/qemu.go

```diff
@@ -867,6 +937,17 @@ func (q *qemu) bootFromTemplate() error {
 	return q.waitMigration()
 }
 
+func (q *qemu) waitVSockReady() error {
+	select {
+	case <-q.vsockConnect:
+		// receive a connection from agent, that is,
+		// vsock device in guest is ready.
+		return nil
+	case <-time.After(vsockReadyTimeOut):
```

Comment:
Is it necessary to introduce a new timeout?  This timeout includes guest kernel boot and kata-agent startup, so 2 seconds might be a little short on a heavily loaded host.

I think a timeout isn't needed here since the container runtime fails container startup after some time anyway.